agent/config.go

```diff
@@ -131,6 +131,10 @@ type DNSConfig struct {
 
 // HTTPConfig is used to fine tune the Http sub-system.
 type HTTPConfig struct {
+	// DisableEndpoints is a list of endpoint prefixes to disable in the
+	// HTTP API. Any requests to these will get a 403 response.
+	DisableEndpoints []string `mapstructure:"disable_endpoints"`
```

Comment:
`s/DisableEndpoints/BlockEndpoints/g`

I'm inclined to just name this `Blacklist` but that may conflict with other blacklists in the future, e.g. network ranges, user agents, ...